cmd/agent/app/integrations.go

```diff
@@ -0,0 +1,269 @@
+			return pyPath, errors.New("unable to find pip executable")
+		}
+	}
+
+	return pyPath, nil
+}
+
+func getConstraintsFilePath() (string, error) {
+	here, _ := executable.Folder()
+	cPath := filepath.Join(here, relConstraintsPath)
+
+	if _, err := os.Stat(cPath); err != nil {
+		if os.IsNotExist(err) {
+			return cPath, err
+		}
+	}
+
+	return cPath, nil
+}
+
+func getTUFConfigFilePath() (string, error) {
+	if _, err := os.Stat(tufConfig); err != nil {
+		if os.IsNotExist(err) {
+			return tufConfig, err
+		}
+	}
+
+	return tufConfig, nil
+}
+
+func tuf(args []string) error {
+	if !allowRoot && !authorizedUser() {
+		return errors.New("Please use this tool as the agent-running user")
+	}
+
+	if flagNoColor {
+		color.NoColor = true
+	}
+
+	err := common.SetupConfig(confFilePath)
+	if err != nil {
+		fmt.Printf("Cannot setup config, exiting: %v\n", err)
+		return err
+	}
+
+	pipPath, err := getCommandPython()
+	if err != nil {
+		return err
+	}
+	tufPath, err := getTUFConfigFilePath()
+	if err != nil && !withoutTuf {
+		return err
+	}
+
+	cmd := args[0]
+	implicitFlags := args[1:]
+	args = append([]string{"-mpip"}, cmd)
+
+	// Add pip power-user flags
+	// cmd-flags go before the actual command
+	cmdFlags, err := tufCmd.Flags().GetStringSlice("cmd-flags")
+	if err == nil {
+		args = append(args, cmdFlags...)
+	}
+	args = append(args, implicitFlags...)
+
+	// Proxy support
+	proxies, err := config.GetProxies()
+	if err == nil && proxies != nil {
+		proxyFlags := fmt.Sprintf("--proxy=%s", proxies.HTTPS)
+		args = append(args, proxyFlags)
+	}
+
+	// idx-flags go after the command and implicit flags
+	idxFlags, err := tufCmd.Flags().GetStringSlice("idx-flags")
+	if err == nil {
+		args = append(args, idxFlags...)
+	}
+
+	tufCmd := exec.Command(pipPath, args...)
+
+	if !withoutTuf {
+		tufCmd.Env = append(os.Environ(),
+			fmt.Sprintf("TUF_CONFIG_FILE=%s", tufPath),
+		)
+	}
+
+	// forward the standard output to the Agent logger
+	stdout, err := tufCmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	go func() {
+		in := bufio.NewScanner(stdout)
+		for in.Scan() {
+			fmt.Println(in.Text())
+		}
+	}()
+
+	// forward the standard error to the Agent logger
+	stderr, err := tufCmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+	go func() {
+		in := bufio.NewScanner(stderr)
+		for in.Scan() {
+			fmt.Println(color.RedString(in.Text()))
+		}
+	}()
+
+	err = tufCmd.Start()
+	if err != nil {
+		fmt.Printf("error running command: %v", err)
+		return err
+	}
+
+	err = tufCmd.Wait()
+	return err
+}
+
+func installTuf(cmd *cobra.Command, args []string) error {
+	constraintsPath, err := getConstraintsFilePath()
+	if err != nil {
+		return err
+	}
+
+	cachePath, err := getTUFPipCachePath()
+	if err != nil {
+		return err
+	}
+
+	tufArgs := []string{
+		"install",
+		"--cache-dir", cachePath,
+		"-c", constraintsPath,
+	}
+
+	tufArgs = append(tufArgs, args...)
+	if !withoutTuf {
+		tufArgs = append(tufArgs, "--index-url", tufPyPiServer)
```

Comment:
With pip+TUF, we don't need both the `--index-url` and `--extra-index-url` options :) I highly recommend removing them...

Reply:
So I needed this to get the install flow to work originally, and more specifically when attempting to deal with the third-party dependencies, etc. If it's not necessary anymore I'll remove them 👍 

That said, I think we have to be careful of how we integrate with the tool (`pip`). Are there security concerns to break the behavior for the switches? Do we not want to allow our customers to specify a PyPI mirror if they have one (maybe that'd be a case where `--extra-index-url` would come in handy)? If we do break switches when the `TUF_CONFIG_FILE` environment variable is specified we should be explicit about this and print a message. The `pip` help should maybe also detail this. My fear is that we end up having so much implicit behavior the tool will might become more painful to maintain (speaking about both the tooling and our `datadog-pip` as well). Anyway, those are questions for a different repo, so this is probably not the right forum to engage in a lengthy discussion regarding those issues.